storage/remote/read_handler.go

```diff
@@ -45,11 +46,17 @@ type readHandler struct {
 // writes them to the provided queryable.
 func NewReadHandler(logger log.Logger, r prometheus.Registerer, queryable storage.SampleAndChunkQueryable, config func() config.Config, remoteReadSampleLimit, remoteReadConcurrencyLimit, remoteReadMaxBytesInFrame int) http.Handler {
 	h := &readHandler{
-		logger:                    logger,
-		queryable:                 queryable,
-		config:                    config,
-		remoteReadSampleLimit:     remoteReadSampleLimit,
-		remoteReadGate:            gate.New(remoteReadConcurrencyLimit),
+		logger:                logger,
+		queryable:             queryable,
+		config:                config,
+		remoteReadSampleLimit: remoteReadSampleLimit,
+		remoteReadGate: gate.New(remoteReadConcurrencyLimit, promauto.With(r).NewCounter(prometheus.CounterOpts{
+			Namespace:   "prometheus",
+			Subsystem:   "api",
+			Help:        "Total sum of how long was the wait at the gate before query execution",
+			Name:        "gate_total_wait_duration_seconds",
```

Comment:
The structure of the code looks good now, but given that we are passing the metric into the gate, we can now also change the metric name and its description to be more specific to how the gate's usage. We have some precedent at https://github.com/prometheus/prometheus/blob/f55ab2217984770aa1eecd0f2d5f54580029b1c0/storage/remote/read_handler.go#L56-L61 for a remote-read-related metric already, so how about `prometheus_api_remote_read_wait_duration_seconds_total` (for the fully expanded metric name), with a description of `Total number of seconds spent waiting at the query gate before remote read query execution`?